test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/FormatterTest.scala

```diff
@@ -758,6 +769,182 @@ class FormatterTest {
         1, 1)
     expectFormatterThrows(classOf[MissingFormatArgumentException], "%10$d", 1)
   }
+
+  /** Tests scenarios where there are multiple errors at the same time, one of
+   *  them being that the conversion is unknown, to make sure that the right
+   *  one takes precedence.
+   */
+  @Test def formatExceptionPrecedenceForUnknownConversionTest_Issue4352(): Unit = {
+    /* In decreasing order of precedence:
+     *
+     * 1. DuplicateFormatFlagsException
+     * 2. UnknownFormatConversionException
+     * 3. Everything else (never happens)
+     */
+
+    // 1-2 DuplicateFormatFlagsException > UnknownFormatConversionException
+    expectFormatterThrows(classOf[DuplicateFormatFlagsException], "%,,j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%,j", 5)
+
+    // 2-3 UnknownFormatConversionException > everything else
+
+    // MissingFormatWidthException
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%-j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%-j", null)
+
+    // IllegalFormatWidthException
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%5j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%5j", null)
+
+    // IllegalFormatFlagsException
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%, j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%, j", null)
+
+    // IllegalFormatPrecisionException
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%.3j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%.3j", null)
+
+    // FormatFlagsConversionMismatchException
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%#j", 5)
+    expectFormatterThrows(classOf[UnknownFormatConversionException], "%#j", null)
```

Comment:
I do not fully understand why the tests here deviate form the pattern:

1. Expect exception that has precedence
2. Remove that error in the pattern and check the other exception is thrown.

IMO this pattern is extremely helpful to make sure the test suite actually tests something.

Reply:
To me it didn't make sense, because I'd have to use a known conversion to provoke the issue, and the test is about the precedence for unknown conversions. But I guess that's fine. I've added additional tests to fit the pattern.